x-pack/packages/kbn-ecs-data-quality-dashboard/impl/data_quality/data_quality_panel/index_properties/helpers.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  IndicesGetMappingIndexMappingRecord,
+  MappingProperty,
+} from '@elastic/elasticsearch/lib/api/types';
+import { sortBy } from 'lodash/fp';
+
+import {
+  getEnrichedFieldMetadata,
+  getFieldTypes,
+  getMissingTimestampFieldMetadata,
+  getPartitionedFieldMetadata,
+} from '../../helpers';
+import type { EcsMetadata, PartitionedFieldMetadata, UnallowedValueCount } from '../../types';
+
+export const ALL_TAB_ID = 'allTab';
+export const ECS_COMPLIANT_TAB_ID = 'ecsCompliantTab';
+export const CUSTOM_TAB_ID = 'customTab';
+export const INCOMPATIBLE_TAB_ID = 'incompatibleTab';
+export const SUMMARY_TAB_ID = 'summaryTab';
+
+export const EMPTY_METADATA: PartitionedFieldMetadata = {
+  all: [],
+  ecsCompliant: [],
+  custom: [],
+  incompatible: [],
+};
+
+export const getSortedPartitionedFieldMetadata = ({
+  ecsMetadata,
+  loadingMappings,
+  mappingsProperties,
+  unallowedValues,
+}: {
+  ecsMetadata: Record<string, EcsMetadata> | null;
+  loadingMappings: boolean;
+  mappingsProperties: Record<string, MappingProperty> | null | undefined;
+  unallowedValues: Record<string, UnallowedValueCount[]> | null;
+}): PartitionedFieldMetadata | null => {
+  if (loadingMappings || ecsMetadata == null || unallowedValues == null) {
+    return null;
+  }
+
+  if (mappingsProperties == null) {
+    return {
+      ...EMPTY_METADATA,
+      incompatible: [getMissingTimestampFieldMetadata()],
```

Comment:
maybe a silly question, but why do we default to showing timestamp as incompatible rather than not showing anything at all?